Modernize using-launchpadlib.rst (#451)
canonical/launchpad-manual: docs/user/how-to/launchpadlib/using-launchpadlib.rst

Reviewer: ```suggestion
.. note::
   ``login_anonymously()`` is only available with launchpadlib version 1.5.4 or higher
   
```
This warning was hidden at the end of this section but should be one of the first things readers see before working through the other steps

Reviewer: ```suggestion
.. note::
If this code complains that 'staging' isn't a URL, it means you're not running
a recent version of launchpadlib. To make it work in Ubuntu 9.10's 
launchpadlib, replace 'staging' with ``launchpadlib.launchpad.STAGING_SERVICE_ROOT``.
```

Get started with launchpadlib
=============================

.. include:: /includes/important_not_revised_help.rst

This document shows how to use a Python client to read and write Launchpad's
data using the launchpadlib library. It doesn't cover the HTTP requests and
responses that go back and forth behind the scenes: for that, see
:ref:`the "hacking" document <developing-the-launchpad-project>`. This
document also doesn't cover the full range of what's possible with Launchpad's
web service: for that, see the
`web service reference documentation <http://launchpad.net/+apidoc/>`_. Check
out the :ref:`API examples page <use-the-python-api>` if you would like to see
more sample code.

Installation
------------

.. tab-set::

   .. tab-item:: Ubuntu

      If you have the latest version of Ubuntu then to install the launchpadlib
      available in the Ubuntu repositories, open a terminal and run the command:

      ::

           sudo apt-get install python3-launchpadlib

      And you are done!

      If you have an older version of Ubuntu then parts of the instructions below may
      not work with the version from the repositories but you should be able to

>       install the latest version of launchpadlib manually.

Reviewer: ```suggestion
      If you have an older version of Ubuntu, some parts of the instructions below may
      not work with the version from the repositories. However, you should be able to
      install the latest version of launchpadlib manually.
```

   .. tab-item:: pip

      On any platform with a working Python and pip, you should be able to

>       say:

Reviewer: ```suggestion
      On any platform with Python and pip, you can install launchpadlib
      using the `pip` command:
```

       from launchpadlib.launchpad import Launchpad
       launchpad = Launchpad.login_anonymously('just testing', 'production', cachedir, version='devel')

- The first argument to ``Launchpad.login_anonymously()`` is a string that
  identifies the web service client. We use this string to gauge client
  popularity and detect buggy or inefficient clients. Here, though, we're just
  testing.

- The second argument tells launchpadlib which Launchpad instance to run against.
  Here, we're using ``production``, which is mapped to the web service root on the
  production Launchpad server: `<https://api.launchpad.net/>`_. Anonymous access
  cannot change the Launchpad dataset, so there's no concern about a bad test
  program accidentally overwriting data. (If you want to play it safe, you could
  use 'staging' instead - though staging is sometimes down for extended periods.)

- The ``version`` argument specifies the API version to use. For historical
  reasons the default is ``'1.0'``, and you may want to use this in certain

>   special circumstances, but in most cases you should use ``'devel'`` so that you

Reviewer: ```suggestion
  circumstances, but in most cases you should use ``'devel'`` so that you
```

> credential that expired), launchpadlib will guide you through authorizing a new

Reviewer: ```suggestion
If you have an existing desktop-wide Launchpad credential, launchpadlib will
find it and use it. If there's no existing desktop credential, launchpadlib will 
guide you through authorizing a new
```
